src/main/java/thaumicboots/api/ItemElectricBoots.java

```diff
@@ -0,0 +1,178 @@
+package thaumicboots.api;
+
+import java.util.List;
+
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DamageSource;
+import net.minecraft.world.World;
+import net.minecraftforge.common.ISpecialArmor;
+
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+import emt.util.EMTConfigHandler;
+import ic2.api.item.ElectricItem;
+import ic2.api.item.IElectricItem;
+import thaumicboots.main.utils.TabThaumicBoots;
+
+public class ItemElectricBoots extends ItemBoots implements IElectricItem, ISpecialArmor {
+
+    public int maxCharge;
+    public int energyPerDamage;
+    public double transferLimit;
+    public boolean provideEnergy;
+
+    public double damageAbsorptionRatio;
+
+    public ItemElectricBoots(ArmorMaterial par2EnumArmorMaterial, int par3, int par4) {
+        super(par2EnumArmorMaterial, par3, par4);
+        setCreativeTab(TabThaumicBoots.tabThaumicBoots);
+        setUnlocalizedName(unlocalisedName);
+    }
+
+    protected void setBootsData() {
+        maxCharge = 0;
+        energyPerDamage = 0;
+        provideEnergy = false; // doesn't need to exist, but someone could make boots that function like a battery
+        transferLimit = 0;
+        damageAbsorptionRatio = 0.5D;
+    }
+
+    public int getEnergyPerDamage() {
+        return energyPerDamage;
+    }
+
+    @Override
+    public boolean canProvideEnergy(ItemStack itemStack) {
+        return provideEnergy;
+    }
+
+    @Override
+    public double getMaxCharge(ItemStack itemStack) {
+        return maxCharge;
+    }
+
+    @Override
+    public double getTransferLimit(ItemStack itemStack) {
+        return transferLimit;
+    }
+
+    // For some reason, sometimes the EMTConfigHandler returns this as 0,
+    // but only when it's called outside of EMT,
+    // this ensures this never happens internally.
+    public float getEMTNanoSpeed() {
+        if ((float) EMTConfigHandler.nanoBootsSpeed == 0.0F) {
+            return 0.275F;
+        }
+        return (float) EMTConfigHandler.nanoBootsSpeed;
+    }
+
+    public float getEMTQuantumSpeed() {
+        if ((float) EMTConfigHandler.quantumBootsSpeed == 0.0F) {
```

Comment:
Why are you casting to float?

Reply:
Bad EMT codebase basically

Reply:
fixed, I don't use EMTConfigHandler anymore, it makes the mod backwards incompatible, and until EMT's GT dep is made optional, this is not acceptable.